app/src/main/kotlin/com/wire/android/media/PingRinger.kt

```diff
@@ -0,0 +1,90 @@
+package com.wire.android.media
+
+import android.content.Context
+import android.content.Context.VIBRATOR_SERVICE
+import android.media.AudioAttributes
+import android.media.AudioManager
+import android.media.MediaPlayer
+import android.os.Build
+import android.os.VibrationEffect
+import android.os.Vibrator
+import android.os.VibratorManager
+import com.wire.android.appLogger
+import javax.inject.Inject
+import javax.inject.Singleton
+
+@Singleton
+class PingRinger @Inject constructor(private val context: Context) {
+
+    private var mediaPlayer: MediaPlayer? = null
+    private var vibrator: Vibrator? = null
+
+    init {
+        val am = context.getSystemService(Context.AUDIO_SERVICE) as AudioManager?
+        am?.isSpeakerphoneOn = false
+        am?.mode = AudioManager.MODE_NORMAL
+        initVibrator()
+    }
+
+    private fun initVibrator() {
+        vibrator = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.S) {
+            val vibratorManager = context.getSystemService(Context.VIBRATOR_MANAGER_SERVICE) as VibratorManager
+            vibratorManager.defaultVibrator
+        } else {
+            @Suppress("DEPRECATION")
+            context.getSystemService(VIBRATOR_SERVICE) as Vibrator
+        }
+    }
+
+    private fun createMediaPlayer(resource: Int) {
+        mediaPlayer = MediaPlayer.create(
+            context,
+            resource,
+            AudioAttributes.Builder()
+                .setContentType(AudioAttributes.CONTENT_TYPE_SONIFICATION)
+                .setUsage(AudioAttributes.USAGE_NOTIFICATION)
+                .build(),
+            0
+        )
+        mediaPlayer?.isLooping = false
+    }
+
+    fun ping(
+        resource: Int,
+        isReceivingPing: Boolean = true
+    ) {
+        vibrateIfNeeded(isReceivingPing)
+        createMediaPlayer(resource)
+        mediaPlayer?.start()
+    }
+
+
+    @Suppress("NestedBlockDepth")
+    private fun vibrateIfNeeded(isReceivingPing: Boolean) {
+        if (isReceivingPing) {
+            vibrator?.let {
+                if (!it.hasVibrator()) {
+                    appLogger.i("Device does not support vibration");
+                    return
+                }
+                val audioManager = context.getSystemService(Context.AUDIO_SERVICE) as AudioManager?
```

Comment:
so what was the `am` object defined in the `init` block for?

Reply:
I based on `CallRinger` to be sure that Ping will work similar to call sound